plugins/woocommerce/tests/php/includes/data-stores/class-wc-product-data-store-cpt-test.php

```diff
@@ -66,4 +66,54 @@ public function test_rating_counts_are_summed_correctly(): void {
 		$product = wc_get_product( $product->get_id() );
 		$this->assertEquals( 10, $product->get_rating_count(), 'The product rating count is the expected value.' );
 	}
+
+	/**
+	 * Test that only one product is created with a unique SKU
+	 * during concurrent requests.
+	 *
+	 * Throw error when two concurrent requests try to create a product with the same SKU.
+	 *
+	 * @return void
+	 */
+	public function test_create_product_with_unique_sku_on_concurrent_requests() {
+		try {
+			$this->create_products_concurrently();
+		} catch ( Exception $e ) {
+			$this->assertMatchesRegularExpression(
```

Comment:
note that this test will not fail even if the exception was not thrown, since the control will never get in this block. consider using the `finally` block, or perhaps `$this->expectException` would be the simplest.